src/app/main.cpp

```diff
@@ -773,21 +778,30 @@ int main( int argc, char *argv[] )
     exit( 1 ); //exit for now until a version of qgis is capabable of running non interactive
   }
 
-  if ( !optionpath.isEmpty() || !configpath.isEmpty() )
-  {
-    // tell QgsSettings to use INI format and save the file in custom config path
-    QSettings::setDefaultFormat( QSettings::IniFormat );
-    QSettings::setPath( QSettings::IniFormat, QSettings::UserScope, optionpath.isEmpty() ? configpath : optionpath );
-  }
-
   // GUI customization is enabled according to settings (loaded when instance is created)
   // we force disabled here if --nocustomization argument is used
   if ( !myCustomization )
   {
     QgsCustomization::instance()->setEnabled( false );
   }
 
-  QgsApplication myApp( argc, argv, myUseGuiFlag, configpath );
+  // Set up the QSettings environment must be done after qapp is created
```

Comment:
Is this comment redundant? It seems contradictory to the order that's happening here.